syntax = "proto3";

package modelservice;

service ModelService {
  rpc GenerateContent (GenerateContentRequest) returns (GenerateContentResponse);
  rpc VerifyTokens (VerifyTokensRequest) returns (VerifyTokensResponse);
  rpc PrepareSpeculative (PrepareSpeculativeRequest) returns (PrepareSpeculativeResponse);
}

message FloatArray {
  repeated float values = 1;
}

message FloatArray2D {
  repeated FloatArray rows = 1;
}

message VerifyTokensRequest {
  string input_text = 1;
  FloatArray2D generated_logits = 2;
}

message VerifyTokensResponse {
  string verified_tokens = 1;
}

message PrepareSpeculativeRequest {
  string input_text = 1;
}

message PrepareSpeculativeResponse {
  string first_tokens = 1;
}

message GenerateContentRequest {
  string prompt = 1;  // 输入提示，可以为空字符串
}

message GenerateContentResponse {
  string generated_text = 1;  // 服务器生成的文本
}
